source/extensions/filters/common/ext_authz/ext_authz_http_impl.cc

```diff
@@ -310,22 +314,47 @@ ResponsePtr RawHttpClientImpl::toResponse(Http::ResponseMessagePtr message) {
     return std::make_unique<Response>(errorResponse());
   }
 
+  // Extract headers-to-remove from the storage header coming from the
+  // authorization server.
+  const auto& storage_header_name = Http::Headers::get().EnvoyAuthHeadersToRemove;
+  // If we are going to construct an Ok response we need to save the
+  // headers_to_remove in a variable first.
+  std::vector<Http::LowerCaseString> headers_to_remove{};
```

Comment:
HeaderVector? Probably we don't need to initialize this here? I mean, we should do this only when we're sure that we have `EnvoyAuthHeadersToRemove` and  it is an OK response.

Reply:
Fixing to avoid initializing.